### ISSUE TITLE
Move to using fly.io machines

### ISSUE THREAD

**Issue by sansaid** [make-or-break-it/promptu#53]
Machines are likely ready now for our use case, so we should start using them

**Comment by sansaid** [comment]
Probably the second time I opened this, but here's what I found so far:

I had a look at fly machines this evening and I don’t think it’s appropriate to use it yet for the web applications. It’s probably only best to apply it to the Zookeeper and the Kafka apps, but even then I’m not so sure. [The docs](https://fly.io/docs/machines/guides-examples/terraform-machines/) seem to suggest that the apps should be deployed alongside a machine (and that the machine configuration replaces the fly.toml). However, I’ve noticed the following things:

1. The machine configuration requires an image in Terraform, but I’m not clear yet how we can pass it a Dockerfile (the CLI accepts a Dockerfile, but not the Terraform API)
2. The machine configuration misses out a lot of config that we can tweak with the fly.toml
3. It’s not clear from the machine configuration how we would do CI/CD - so in other words, if we change the app code, how do we tell it to deploy the latest changes of the image?
4. The docs seem to suggest that if [the machine](https://fly.io/docs/machines/guides-examples/machines-app-using-flyctl/#differences-from-a-nomad-app) goes down, they don’t bring it back up elsewhere (???)

Over all, I think it’s best we keep to using the apps as they are, even if [fly.io](http://fly.io/) say that they’ll be deprecated at some point. My personal opinion at this stage is that machines seem to new to use stably.
